test/test_fx.py

```diff
@@ -1419,6 +1421,49 @@ def test_copy_it(self):
         self.assertEqual(d, deepcopy(d))
         self.assertEqual(l, deepcopy(l))
 
+    def test_get_torch_func_signature(self):
+        for key in dir(torch):
+            obj = getattr(torch, key)
+            if callable(obj):
+                schemas = get_signature_for_torch_op(obj)
+
+    def test_get_torch_func_signature_exhaustive(self):
```

Comment:
@Lilyjjo @mruberry Can someone take a look at this? At a high level, I'm trying to iterate over as many `torch.` namespace ops as I can and get sample inputs to each of them. However, there's a bunch that fail (see the exclusion list below) and of the ones that succeed there only seem to be ~100 of them? Is there another way I should be getting exhaustive operator tests?

Reply:
I suggest implementing this using the @ops decorator like this:

https://github.com/pytorch/pytorch/blob/aeb3e933515672a61cb21498fdbdef80a46ae0ee/test/test_ops.py#L447

which will instantiate one test per op and only test the floating point dtype. You can further limit the test to only run on the CPU, for example, by adding the @onlyCPU decorator:

```
@onlyCPU
@ops(op_db, allowed_dtypes=(torch.float,))
def test_get_torch_func_signature_exhaustive(self, device, dtype, op):
```

To use the @ops decorator you'll have to put this test into a device generic framework test suite, but that's just a single call:

https://github.com/pytorch/pytorch/blob/aeb3e933515672a61cb21498fdbdef80a46ae0ee/test/test_ops.py#L603

One reason this might be helpful is when someone adds a new OpInfo this will test will run and if it fails you can inform the PR's author what they should do. In this case you likely want author's to skip the test, and they can add a skip to the OpInfo:

https://github.com/pytorch/pytorch/blob/aeb3e933515672a61cb21498fdbdef80a46ae0ee/torch/testing/_internal/common_methods_invocations.py#L1480

This is probably a little easier than working through a custom skip method.

You could also implement a custom filter for all LAPACK operations by testing for the @skipCPUIfNoLapack decorator in an OpInfo's "decorators" attribute. Example:

https://github.com/pytorch/pytorch/blob/aeb3e933515672a61cb21498fdbdef80a46ae0ee/torch/testing/_internal/common_methods_invocations.py#L1695

That will save you having to write a lot of the skips here.

As for coverage you're right that OpInfos don't currently cover every operator. @zdevito was also asking about operator coverage earlier today. Let me sync with our Quansight partners to prioritize OpInfo creation. cc @rgommers @mattip

Reply:
Thanks @mruberry! I probably wouldn't have figured that out myself; it's a bunch of declarative pieces that have to fit together when my mental model (and the status quo before iirc) was "there's a structure I can loop over to get info for all the ops)

BTW it seems that the tests fail if the op isn't invoked within the test implementation, like so:

```
======================================================================
FAIL: test_get_torch_func_signature_exhaustive_floor_divide_cpu_float32 (__main__.TestOperatorSignaturesCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/users/jamesreed/pytorch/torch/testing/_internal/common_device_type.py", line 287, in instantiated_test
    result = test_fn(self, *args)
  File "/data/users/jamesreed/pytorch/torch/testing/_internal/common_utils.py", line 2095, in inner
    fn(self, *args, **kwargs)
  File "/home/jamesreed/local/miniconda3/lib/python3.7/contextlib.py", line 119, in __exit__
    next(self.gen)
  File "/data/users/jamesreed/pytorch/torch/testing/_internal/common_utils.py", line 1396, in assertWarnsOnceRegex
    self.fail('no warning caught')
AssertionError: no warning caught
```

This is fine since I decided to add an op invocation anyway (I was originally just going to bind schema and leave it at that) but FYI

Reply:
Let me know if you have questions. The OpInfo pattern is updating quickly and [our documentation for it](https://github.com/pytorch/pytorch/wiki/Writing-tests-in-PyTorch-1.8) is pretty minimal and already out of date.

Does that warning only appear for floor_divide? It might be because it's wrapping the test with

https://github.com/pytorch/pytorch/blob/0c8f16622bf6f410b67adec08c15692d5c404225/torch/testing/_internal/common_methods_invocations.py#L2104

Maybe when floor_divide is run using fx the value of flag controlling eager warn once behavior isn't respected? It's a global flag. See:

https://pytorch.org/docs/master/generated/torch.set_warn_always.html?highlight=warn%20once

A more complete filtering option could involve deep copying the op_db iterable, removing the ops that require LAPACK (per above) and removing this decorator from ops like floor_divide. Then you could pass the filtered/edited iterable of ops to the @ops decorator. For example, here's a simple filter that only instantiates tests for ops with aliases:

https://github.com/pytorch/pytorch/blob/0c8f16622bf6f410b67adec08c15692d5c404225/test/test_ops.py#L351